packages/components/file-uploader/src/file-uploader.css

```diff
@@ -0,0 +1,421 @@
+    width: calc(15 * var(--ts-unit-half));
+  }
+  100% {
+    left: 150%;
+    width: 0;
+  }
+}
+
+:host > .file-card-wrapper > .file-card--uploading label {
+  font-weight: 600;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+:host > .file-card-wrapper .icon:last-of-type {
+  background-color: var(--ts-color-gray-dark);
+  height: var(--ts-unit);
+  margin-right: var(--ts-unit-half);
+  width: var(--ts-unit);
+}
+
+:host > .file-card-wrapper .icon:last-of-type:hover {
+  cursor: pointer;
+}
+
+:host > .file-card-wrapper .icon:last-of-type > span {
+  margin-top: -1px;
+}
+
+/* Failed state..................................................... */
+
+:host > .file-card-wrapper > .file-card--failed {
+  border-color: #FF1744;
+}
+
+:host > .file-card-wrapper > .file-card--failed:hover {
+  border-color: #FF1744;
+}
+
+:host > .file-card-wrapper > .file-card--failed .icon:first-of-type {
+  background-color: #FF1744;
+  border-left: 14px solid var(--ts-color-white);
+  border-right: 14px solid var(--ts-color-white);
+  border-bottom: 28px solid red;
+  border-radius: 0px;
+}
+
+:host > .file-card-wrapper > .file-card--failed .icon:first-of-type > span {
+  margin-top: 34px;
+}
+
+:host > .file-card-wrapper > .file-card--failed .file-upload--info {
+  display: flex;
+  flex-direction: row;
+  font-size: calc(1px + var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card--failed small:last-of-type {
+  color: var(--ts-color-blue);
+  cursor: pointer;
+  margin-left: calc(0.2 * var(--ts-unit-half));
+}
+
+/* Download state.................................................. */
+
+:host > .file-card-wrapper > .file-card--download:hover {
+  border-color: var(--ts-color-blue);
+  cursor: pointer;
+}
+
+:host > .file-card-wrapper > .file-card--download .icon:first-of-type {
+  background-color: var(--ts-color-white);
+  color: var(--ts-color-slate);
+  height: 44px;
+  position: relative;
+  transition: color 0.15s ease;
+  width: 35px;
+}
+
+:host > .file-card-wrapper > .file-card--download:hover .icon:first-of-type {
+  color: var(--ts-color-blue);
+}
+
+:host > .file-card-wrapper > .file-card--download .icon:first-of-type span {
+  bottom: 4px;
+  color: var(--ts-color-black);
+  font-size: calc(1px + var(--ts-unit-half));
+  position: absolute;
+  text-transform: uppercase;
+  transition: color 0.15s ease;
+}
+
+:host > .file-card-wrapper > .file-card--download:hover .icon:first-of-type span {
+  color: var(--ts-color-blue);
+}
+
+:host > .file-card-wrapper > .file-card--download:hover .icon:first-of-type span:before {
+  animation-name: appear;
+  animation-duration: 0.15s;
+  animation-iteration-count: 1;
+  animation-timing-function: linear;
+  background-color: var(--ts-color-blue);
+  border-radius: 50%;
+  content: '';
+  height: 22px;
+  left: 15px;
+  position: absolute;
+  top: 8px;
+  width: 22px;
+}
+
+:host > .file-card-wrapper > .file-card--download:hover .icon:first-of-type span:after {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Simplify this over-specified selector. Maximum allowed depth: 3. Actual depth: 5 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aoverspecific-selectors)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this usage of the experimental "host" pseudo. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aexperimental-pseudo-usage)